Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,45 @@ def installed?
       !versions.empty?
     end
 
+    def outdated?(greedy = false)
+      greedy = true if greedy
+      @outdated ||= {}
+      @outdated[greedy] ||= _check_outdated(greedy)
+      !@outdated[greedy].empty?
+    end
+
+    def outdated_versions(greedy = false)
+      greedy = true if greedy
+      @outdated ||= {}
+      @outdated[greedy] ||= _check_outdated(greedy)
```

Comment:
`#outdated?` and `outdated_versions` are both called when `--verbose`. This is why the information is cached (`versions` is expensive).

Is it ok to (drop caching and the `#outdated_versions` method and) let `#outdated?` return a boolean **and** the list of outdated versions?